website/project/decorators.py

```diff
@@ -229,6 +243,7 @@ def wrapped(*args, **kwargs):
 # Create authorization decorators
 must_be_contributor = _must_be_contributor_factory(False)
 must_be_contributor_or_public = _must_be_contributor_factory(True)
+must_be_contributor_or_public_but_not_view_only = _must_be_contributor_factory(include_public=True, include_view_only_anon=False)
```

Comment:
A better name for this might be `must_be_contributor_or_public_but_not_anonymized`, as non-anonymized view-only links should still be be allowed access.